platform/platform-sportpaper/src/main/java/tc/oc/pgm/platform/sportpaper/SpMiscUtil.java

```diff
@@ -83,4 +89,19 @@ public ThrownPotion spawnPotion(Location loc, ItemStack item) {
   public double getArrowDamage(Arrow arrow) {
     return arrow.spigot().getDamage();
   }
+
+  @Override
+  public int getWorldDataVersion(Path levelDatPath) {
+    try {
+      var compoundTag =
+          NBTCompressedStreamTools.a(Files.newInputStream(levelDatPath, StandardOpenOption.READ));
+      var dataTag = compoundTag.getCompound("Data");
+      return dataTag.hasKeyOfType("DataVersion", NBT_TAG_ANY_NUMERIC)
+          ? dataTag.getInt("DataVersion")
+          : -1;
+    } catch (Exception ignored) {
```

Comment:
style: might be worthwhile to catch all throwables to keep it consistent with modern
```suggestion
    } catch (Throwable ignored) {
```